mempool/clist_mempool.go

```diff
@@ -52,19 +51,19 @@ type CListMempool struct {
 
 	// Data in the following variables must to be kept in sync and updated atomically.
 	txsMtx    cmtsync.RWMutex
-	lanes     map[types.Lane]*clist.CList     // each lane is a linked-list of (valid) txs
+	lanes     map[types.LaneID]*clist.CList   // each lane is a linked-list of (valid) txs
 	txsMap    map[types.TxKey]*clist.CElement // for quick access to the mempool entry of a given tx
-	laneBytes map[types.Lane]int64            // number of bytes per lane (for metrics)
+	laneBytes map[types.LaneID]int64          // number of bytes per lane (for metrics)
 	txsBytes  int64                           // total size of mempool, in bytes
 	numTxs    int64                           // total number of txs in the mempool
 
-	addTxChMtx    cmtsync.RWMutex      // Protects the fields below
-	addTxCh       chan struct{}        // Blocks until the next TX is added
-	addTxSeq      int64                // Helps detect is new TXs have been added to a given lane
-	addTxLaneSeqs map[types.Lane]int64 // Sequence of the last TX added to a given lane
+	addTxChMtx    cmtsync.RWMutex        // Protects the fields below
+	addTxCh       chan struct{}          // Blocks until the next TX is added
+	addTxSeq      int64                  // Helps detect is new TXs have been added to a given lane
+	addTxLaneSeqs map[types.LaneID]int64 // Sequence of the last TX added to a given lane
 
 	// Immutable fields, only set during initialization.
-	defaultLane types.Lane
+	defaultLane string
 	sortedLanes []types.Lane // lanes sorted by priority, in descending order
```

Comment:
Just realised that `sortedLanes` is used only by iterators so we could move it inside the iterators and create it when iterators are created. In any case, I would split this array in two: one with sorted lane ids and a map from lane id to priority, so then we can get rid of the `types.Lane` struct.

Reply:
I liked this suggestion but when I was looking at removing it and having the sort happen every time we make an iterator, I am not sure it is the best way to go. The iterators are created a lot (on every reap etc.) and we'd have to sort every time. 

I also looked at removing the struct, I can do it but in many places we have to create two arrays that we have to iterator over in parallel instead of one. I personally find this cleaner, but if you feel strongly about it I can remove it?